Merge back 'chore_release-pd-8.7.0' into 'edge'

ddcc4 opened the pull request:
This pulls in Nick's latest fixes for PD 8.7.0.

I had to fix a merge conflict between @ncdiehl's PR #20338 and my PR #20326 in `safePipetteMovements.ts`

ncdiehl11 committed:
fix(step-generation): add tip overlaps in collision logic (#20338)

Previously, step-generation's collision detection logic hard-set the
overlap of the potential picked-up tip on the nozzle to 0mm (gulp, my
bad!). This PR pulls in the actual tip overlap to accurately determine
the pipette bounds' z value taking into account this overlap. This
protects us from under-estimating how low the pipette nozzles will be
off the deck, and correctly detecting collisions that are within ~10mm.

ncdiehl11 committed:
fix(step-generation): fix OT-2 collision logic (#20340)

This PR fixes logic for determining module heights for OT-2 modules.
This allows us to correctly detect possible collisions with surroudning
modules during partial tip pipetting on OT-2.

koji committed:
fix(workflow):  pd workflow for sourcemap (#20282)

* fix(workflow):  pd workflow for sourcemap

ncdiehl11 reviewed (approved):
step generation looks good. thanks 💃

ncdiehl11 committed:
fix(step-generation, shared-data): wire up deck extent checks in safePipetteMovements (#20345)

This PR wires up deck extent checks in `safePipetteMovements.ts` in step
generation. In short, we need to check not only on-deck collisions with
partial tip configuration pipette movements, but also absolute pipette
target positions relative to the limits of the gantry. As part of this
fix, I also add a thermocycler offset to correctly get the well position
for wellplates inside of a thermocycler.

ddcc4 commented:
FYI, since people ended up changing `edge` before I could get this merged, this mergeback will now include @ncdiehl11's latest PD PR as well :)

ddcc4 committed:
Merge back 'chore_release-pd-8.7.0' into 'edge' (#20346)

Resolved conflicts:
- step-generation/src/utils/safePipetteMovements.ts
